listenbrainz/webserver/decorators.py

```diff
@@ -45,3 +46,34 @@ def wrapped_function(*args, **kwargs):
         f.provide_automatic_options = False
         return update_wrapper(wrapped_function, f)
     return decorator
+
+
+def api_listenstore_needed(func):
+    """
+        This API decorator checks to see if timescale is online (by having
+        a DB URI) and if not, it raises APIServiceUnavailable.
+    """
+    @wraps(func)
+    def decorator(*args, **kwargs):
+        from listenbrainz.webserver.errors import APIServiceUnavailable
+        if not current_app.config["SQLALCHEMY_TIMESCALE_URI"]:
```

Comment:
we don't need to make this change for today's release, but I think that this check would be better to verify if the timescale listenstore is None instead of looking at the config option. that way we can support bringing up LB with these error messages even if there is a URI set but the host is down (for example). I'll open a ticket after the review.

Reply:
done